src/sweeps/bayes_search.py

```diff
@@ -407,19 +407,23 @@ def _construct_gp_data(
 
     X_norms = params.normalize_runs_as_array(runs)
     worst_metric = impute(goal, metric_name, ImputeStrategy.worst, runs=runs)
+
+    def get_metric(strategy: ImputeStrategy):
```

Comment:
can we add a short comment describing why this function is needed and why we precalculate worst_metric above? (its for performance, since its the same for all runs in `runs` its better to only calculate it once than recalculate it for every iteration of the for loop below)